app/screens/forgot_password/forgot_password.test.tsx

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import {renderWithIntl, fireEvent, waitFor} from '@test/intl-test-helper';
+
+import ForgotPassword from './index';
+import * as UserAPICalls from '@requests/remote/user';
+
+describe('ForgotPassword', () => {
+    const baseProps = {
+        componentId: 'ForgotPassword',
+    };
+
+    test('should match snapshot', () => {
+        const {toJSON} = renderWithIntl(<ForgotPassword {...baseProps}/>);
+        expect(toJSON()).toMatchSnapshot();
+    });
+
+    test('Error on failure of email regex', async () => {
+        const {getByTestId} = renderWithIntl(<ForgotPassword {...baseProps}/>);
+        const emailTextInput = getByTestId('forgot.password.email');
+        const resetButton = getByTestId('forgot.password.button');
+
+        fireEvent.changeText(emailTextInput, 'bar');
+
+        await waitFor(() => {
```

Comment:
If the above changes the state, this should be `act` instead of `waitFor`, right ?

Reply:
This one works.

Reply:
I know it works.. but from the testing-library this should use `act` not really waitFor.. 0/5 I'll let @josephbaylon comment on this